core/src/main/java/org/elasticsearch/ingest/core/IngestDocument.java

```diff
@@ -492,7 +526,7 @@ private static Object deepCopy(Object value) {
             return copy;
         } else if (value == null || value instanceof String || value instanceof Integer ||
             value instanceof Long || value instanceof Float ||
-            value instanceof Double || value instanceof Boolean) {
+            value instanceof Double || value instanceof Boolean || value instanceof byte[]) {
```

Comment:
I think that we need to deep copy the byte array instead of returning it as-is. Otherwise subsequent changes to the array against a copy of the document will affect all the other copies too.